src/frontend/src/expr/type_inference.rs

```diff
@@ -85,12 +84,9 @@ pub fn infer_type(func_type: ExprType, inputs_type: Vec<DataType>) -> Result<Dat
         DataTypeName::Timestampz => DataType::Timestampz,
         DataTypeName::Time => DataType::Time,
         DataTypeName::Interval => DataType::Interval,
-        DataTypeName::Struct => DataType::Struct {
-            fields: Arc::new([]),
-        },
-        DataTypeName::List => DataType::List {
-            datatype: Box::new(DataType::Int32),
-        },
+        DataTypeName::Struct | DataTypeName::List => {
+            panic!("functions returning struct or list can not be inferred")
```

Comment:
Why?

Reply:
I think our current inferring probably need to be enhanced. Currently, `infer_type_name` looks up function signature (DataTypeName) from a hash map returns an DataTypeName. This could work because most data types do not have type arguments, but this doesn't hold for `Struct` or `List`.

Reply:
I agree it needs to be enhanced. Before we have implemented some functions on list/struct and have a better idea how it should look like, panicking here makes sure developers working on struct/list would not accidentally call into this function.

Better suggestions are welcomed.

Reply:
Yes, so I disallow inferring functions that return list or struct, currently, including ARRAY expression and ROW expression, they should use `FunctionCall::new_unchecked()`.